### PR TITLE
[FLINK-13499][maprfs] Remove dependency on MapR artifact repository

### PR DESCRIPTION
**Pull request by StephanEwen**
## What is the purpose of the change

This change removes the MapR FS connector to work based on reflection and avoid a hard dependency on any of the MapR vendor-specific artifacts. That allows us to get rid of the dependency without regressing on the support for the file system.

The MapR artifact repository causes some problems. Access to the a secure (https://) endpoint is very flaky.

## Verifying this change

Verified by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **No**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **No**
  - The serializers: **No**
  - The runtime per-record code paths (performance sensitive): **No**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: **No**
  - The S3 file system connector: **No**

## Documentation

  - Does this pull request introduce a new feature? **No**
  - If yes, how is the feature documented? **No**

### CODE REVIEW

File: flink-filesystems/flink-mapr-fs/src/test/java/org/apache/flink/runtime/fs/maprfs/MapRNotInClassPathTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * A class with tests that require to be run in a MapR/Hadoop-free environment,
+ * to test proper error handling when no Hadoop classes are available.
+ *
+ * <p>This class must be dynamically loaded in a MapR/Hadoop-free class loader.
+ */
+public class MapRNotInClassPathTest {
```
Comment (tillrohrmann) [inline flink-filesystems/flink-mapr-fs/src/test/java/org/apache/flink/runtime/fs/maprfs/MapRNotInClassPathTest.java]: Missing `extends TestLogger`.

**Comment by StephanEwen** [comment]
Thanks. Will change this and merge...

**Comment by aljoscha** [comment]
Looks good to merge, yes! The build failure is in the python profile

**Comment by zentol** [comment]
@StephanEwen Did you forget to close this PR?

**Pull request closed without merging**